source/agora/consensus/Fee.d

```diff
@@ -401,6 +401,29 @@ public class FeeManager
         } ();
     }
 
+    /***************************************************************************
+
+        Calculate the total fees of a transaction (including the payload fee)
+
+        Params:
+            tx = transaction for which we want to calculate the fee
+            peekUTXO = UTXO finder
+
+        Returns: the total fee of the transaction (including the payload fee)
+
+    ***************************************************************************/
+
+    public Amount getTXFee (in Transaction tx, scope UTXOFinder peekUTXO) nothrow @safe
+    {
+        Amount tot_fee;
+        Amount tot_data_fee;
+
+        getTXSetFees([tx], peekUTXO, tot_fee, tot_data_fee);
+        tot_fee.add(tot_data_fee);
+
+        return tot_fee;
+    }
```

Comment:
Maybe you could just modify the other function to return the `sum`, since it currently returns `void` ?

Reply:
I think it is better to have only 1 clearly defined responsibility for every function and combine them if necessary. Not everyone needs the sum of tot_fee and tot_data_fee, so we should not calculate it unnecessarily(even though this is just an addition).

(In fact, I think the original function should just return an array of tot_fee and tot_data fee and have a 3rd function aggregate it, if aggregation is necessary)